indico/modules/events/abstracts/legacy.py

```diff
@@ -0,0 +1,506 @@
+            if status.getTrack() is not None:
+                abstract.as_legacy.addTrack(status.getTrack())
+            abstract.as_legacy.setCurrentStatus(AbstractStatusSubmitted(abstract.as_legacy))
+            abstract.contribution = None
+
+
+def person_from_data(person_data, event):
+    user = User.find_first(~User.is_deleted, User.all_emails.contains(person_data['email'].lower()))
+    if user:
+        return EventPerson.for_user(user, event)
+
+    person = EventPerson.find_first(email=person_data['email'].lower())
+    if not person:
+        person = EventPerson(event_new=event, **person_data)
+    return person
+
+
+@no_autoflush
+def contribution_from_abstract(abstract, session):
+    from MaKaC.review import AbstractStatusAccepted
+
+    event = abstract.as_new.event_new
+    duration = session.default_contribution_duration if session else timedelta(minutes=15)
+    links = []
+
+    for auth in abstract.getAuthorList():
+        author_type = AuthorType.primary if abstract.isPrimaryAuthor(auth) else AuthorType.secondary
+        person_data = {
+            'title': UserTitle.from_legacy(to_unicode(auth.getTitle())),
+            'email': to_unicode(auth.getEmail()),
+            'address': to_unicode(auth.getAddress()),
+            'affiliation': to_unicode(auth.getAffiliation()),
+            'first_name': to_unicode(auth.getFirstName()),
+            'last_name': to_unicode(auth.getSurName()),
+            'phone': to_unicode(auth.getTelephone())
+        }
+
+        person = person_from_data(person_data, event)
+        person_data.pop('email')
+        links.append(ContributionPersonLink(author_type=author_type, person=person,
+                                            is_speaker=abstract.isSpeaker(auth), **person_data))
+
+    contrib = create_contribution(event, {'title': abstract.getTitle(), 'duration': duration,
+                                          'person_link_data': {link: True for link in links}})
+    contrib.abstract = abstract.as_new
+
+    contrib.description = abstract.as_new.description
+    for field_val in abstract.as_new.field_values:
+        new_val = ContributionFieldValue(contribution_field=field_val.contribution_field, data=field_val.data)
+        contrib.field_values.append(new_val)
+
+    # if this is an accepted abstract, set track and type if present
+    status = abstract.getCurrentStatus()
+    if isinstance(status, AbstractStatusAccepted):
+        track = abstract.as_new.accepted_track
+        contrib_type = abstract.as_new.accepted_type
+        if track:
+            contrib.track_id = int(track.getId())
+        if contrib_type:
+            contrib.type = contrib_type
+
+    if session:
+        contrib.session = session
+
+    return contrib
+
+
+class AbstractFieldWrapper(object):
+    """Wraps an actual ``ContributionField`` in order to emulate the old ``AbstractField``s."""
+
+    @classmethod
+    def wrap(cls, field):
```

Comment:
doesn't matter for a legacy wrapper that will eventually go away, but this looks like a perfect case for `__new__`